sys/net/gnrc/link_layer/lwmac/lwmac.c

```diff
@@ -880,7 +872,16 @@ static void _lwmac_msg_handler(gnrc_netif_t *netif, msg_t *msg)
             lwmac_schedule_update(netif);
             break;
         }
-
+#if (GNRC_MAC_ENABLE_DUTYCYCLE_RECORD == 1)
+        case GNRC_MAC_TYPE_GET_DUTYCYCLE: {
+            /* Output LWMAC's radio duty-cycle ratio */
+            uint64_t duty = (uint64_t) rtt_get_counter();
+            duty = ((uint64_t) netif->mac.prot.lwmac.awake_duration_sum_ticks) * 100 /
+                   (duty - (uint64_t)netif->mac.prot.lwmac.system_start_time_ticks);
+            printf("[LWMAC]: achieved radio duty-cycle: %u %% \n", (uint16_t) duty);
```

Comment:
Why is `duty` cast to `uint16_t` if it was `uint64_t` before?

Reply:
Just to fit the printf in `%u`.

Reply:
Since the result seems to be a percentage I guess this is ok...

Reply:
Maybe go for `unsigned` instead though (you may squash that immediately in).

Reply:
(`%u` is expecting `unsigned` not `uint16_t` and is in this case the more portable cast).